database/migrations/2023_10_25_153643_qs_batches_add_pending_column.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class QsBatchesAddPendingColumn extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('qs_batches', function (Blueprint $table) {
+            $table->timestampTz('pending_since')->nullable()->default(null);
+            $table->unsignedInteger('processing_attempts')->default(0);
```

Comment:
There'd be a million ways to model (extra failures table, single field), but this seemed like the most "dumb" solution to me. If you have a better idea or an opinion on this, now is the time to speak up.

Reply:
In my head `pending`, `failed` and `done` would be possible states of one shared state here - but I can live with it not being 1:1 implemented that way, especially because I see how it makes sense to retro-fit it this way

Reply:
Yeah, I thought about how this would be the cleanest option as well, introducing a new `state` field and then allowing one of the three values in there. As you already surmised I was worried about having to migrate all the existing items and what happens with the ones that are processed while running the migration. I'll give it another think.